Models/TreeModel.cpp

```diff
@@ -48,13 +48,13 @@ QVariant TreeModel::data(const QModelIndex &index, int role) const {
 
     auto it = iconMap.find(item->type_case());
     if (it == iconMap.end()) return ArtManager::GetIcon("info");
-    return it->second;
+    const QIcon &icon = it->second;
+    if (item->type_case() == TypeCase::kFolder && item->child_size() <= 0) {
+      return icon.pixmap(QSize(18, 18), QIcon::Disabled);
```

Comment:
Be mindful of your use of constants. Why 18? What if my display resolution is 4K?

Reply:
Right, 18x18 is the actual size of the folder icon. The problem here is a bit of a shitty API sort of.

`QIcon::actualSize` just returns a size smaller, which is what `QIcon::pixmap` does anyway, so it makes no sense to call it when you can just pass a maximum `QSize(99,99)` to `QIcon::pixmap` if you want to go that route.
http://doc.qt.io/qt-5/qicon.html#actualSize

`QIcon::availableSizes` does give me a way to grab the largest size:
http://doc.qt.io/qt-5/qicon.html#availableSizes

Reply:
Ok, I think with a09b855e3ecad0971d91ec268f8a4d0ce21dc8fc it's a little better. We'll have to come back to this later and look at zoom levels for this tree view anyway, but that's for another time.